third-party/realdds/src/dds-device-server.cpp

```diff
@@ -135,3 +172,33 @@ void dds_device_server::publish_notification( topics::flexible_msg && notificati
     _notification_server->send_notification( std::move( notification ) );
 }
 
+
+void dds_device_server::on_control_message_received()
+{
+    topics::flexible_msg data;
+    eprosima::fastdds::dds::SampleInfo info;
+    while ( topics::flexible_msg::take_next( *_control_reader, &data, &info ) )
+    {
+        if ( !data.is_valid() )
+            continue;
+
+        _control_dispatcher.invoke( [&]( dispatcher::cancellable_timer ) { handle_control_message( data ); } );
+    }
+}
+
+void dds_device_server::handle_control_message( topics::flexible_msg control_message )
+{
+    auto const & j = control_message.json_data();
```

Comment:
`json_data` returns a json object, but not a reference. This should be putting up a warning?

Reply:
gcc issued a warning when I used a non-const reference

Reply:
It shouldn't be a reference.

Reply:
Won't it be more efficient

Reply:
If `int foo() { return 5; }`,
Can you do `int & x = foo();`?
You cannot assign a reference to a temporary variable. If the compiler does not generate a warning, the performance will still be the same.

Reply:
Fixed